kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/consumer/streamable/StreamableKafkaMessageSource.java

```diff
@@ -127,14 +129,29 @@ public BlockingStream<TrackedEventMessage<?>> openStream(TrackingToken trackingT
         return new KafkaMessageStream(buffer, closeHandler);
     }
 
+    @Override
+    public TrackingToken createHeadToken() {
+        return KafkaTrackingToken.newInstance(ConsumerPositionsUtil.getHeadPositions(
+                consumerFactory.createConsumer(null),
```

Comment:
Curious: why don't we want a `Consumer` that's tied to a `group`?

Reply:
In this case, we manage the tokens ourselves, which means we should rely on the segments for partitioning. If we use a consumer group, each one with the same name would only get half of the partitions. Using a group should only be used by the subscribing event processor, where Kafka user consumer groups to divide the stream, and also uses consumer groups to store offsets in Kafka itself.